src/pages/api/sanity/lessons/create.ts

```diff
@@ -13,54 +14,119 @@ type LessonData = {
   }
 }
 
+type SanityReference<SanityType> = {
+  _type: SanityType
+  _ref: string
+}
+
+type SanityReferenceArray<SanityType> = Array<
+  {
+    _key: string
+  } & SanityReference<'reference'>
+>
+
 type SanityVideoResource = {
-  _type: string
+  _type: 'videoResource'
   _id: string
   filename: string
   originalVideoUrl: string
 }
 
 type SanityLesson = {
-  _type: string
+  _type: 'lesson'
+  _id: string
   title: string
   resource: {
     _type: string
     _ref: string
   }
 }
 
+type SanityCourse = {
+  _type: 'course'
+  title: string
+  collaborators: SanityReferenceArray<'collaborator'>
+  lessons: SanityReferenceArray<'lesson'>
+}
+
 const sanityClient = client({
   projectId: 'sb1i5dlc',
   dataset: 'production',
   useCdn: true,
   token: process.env.SANITY_EDITOR_TOKEN,
 })
 
-async function formatSanityMutationForLessons(lessons: LessonData[]) {
+const sanityIdForDocumentType = async (
+  documentType: string,
+): Promise<string> => {
+  const id = await nanoid()
+  return `${documentType}-${id}`
+}
```

Comment:
Ensure nanoids that are generated are valid

Because `-` is in the base alphabet for nanoids, it sometimes generates
an id that starts with `-`. An id starting with `-` is not valid as a
Sanity `_id`. By prepending the document type to the generated id, we
ensure that doesn't happen and add a little helpful metadata to the
`_id` field.